### PR TITLE
on_rm_rf_error: ignore os.open (no warning)

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/pytest-dev/pytest/pull/6044/files#r339321752

### CODE REVIEW

File: src/_pytest/pathlib.py [hunk 1]
```diff
@@ -59,13 +59,14 @@ def on_rm_rf_error(func, path: str, exc, *, start_path) -> bool:
         return False
 
     if func not in (os.rmdir, os.remove, os.unlink):
-        warnings.warn(
-            PytestWarning(
-                "(rm_rf) unknown function {} when removing {}:\n{}: {}".format(
-                    path, func, exctype, excvalue
+        if func not in (os.open,):
```
Comment (RonnyPfannschmidt) [inline src/_pytest/pathlib.py]: shouldn't we look at retrying for a directory there?

Comment (blueyed) [reply src/_pytest/pathlib.py]: IIRC it is not possible to rety just that call.  (see the other comments about this)

Comment (RonnyPfannschmidt) [reply src/_pytest/pathlib.py]: based on https://github.com/python/cpython/blob/a8fb9327fb9387b404d4b6ce55c608125f66b9ae/Lib/shutil.py#L637-L642

we cant retry, this is frickle

Comment (blueyed) [reply src/_pytest/pathlib.py]: My idea initially was to retry the whole operation, not just the function call (IIRC there's an older/closed PR with that attempt).
Currently we would also emit a warning for e.g. `os.lstat` - a warning for "unknown function" in that context also sounds strange.
As of now I think we should have no warnings for unhandled functions, or whitelist all the ones being used.

Comment (RonnyPfannschmidt) [reply src/_pytest/pathlib.py]: thanks for the research

**Comment by blueyed** [comment]
Hypothesis did something!

```
______________________ TestMetafunc.test_idval_hypothesis ______________________
6120[gw0] darwin -- Python 3.7.0 /Users/travis/build/pytest-dev/pytest/.tox/py37-xdist/bin/python
6121
6122self = <metafunc.TestMetafunc object at 0x1054ca518>
6123
6124    @hypothesis.given(strategies.text() | strategies.binary())
6125>   @hypothesis.settings(
6126        deadline=400.0
6127    )  # very close to std deadline and CI boxes are not reliable in CPU power
6128    def test_idval_hypothesis(self, value):
6129E   hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 6 valid examples in 1.03 seconds (0 invalid ones and 2 exceeded maximum size). Try decreasing size of the data you're generating (with e.g.max_size or max_leaves parameters).
6130E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.too_slow to the suppress_health_check settings for this test.
6131
6132testing/python/metafunc.py:195: FailedHealthCheck
6133---------------------------------- Hypothesis ----------------------------------
6134You can add @seed(145025405721631010594444570234322154546) to this test or run pytest with --hypothesis-seed=145025405721631010594444570234322154546 to reproduce this failure.
6135=============================== warnings summary ===============================
6136
```
;) (#4977)

**Comment by blueyed** [comment]
Added two tests (there are two issues).
Re changelog: it fixes issues with ba4b8c869, which was not released.

**Comment by blueyed** [comment]
> a quick unittest

btw: would have been good if the test was more specific / hardened before already.. - then at least one of the issues here would not have popped up.

**Comment by nicoddemus** [comment]
> btw: would have been good if the test was more specific / hardened before already.

Agreed

**Comment by blueyed** [comment]
@nicoddemus 
You might want to backport it (https://github.com/pytest-dev/pytest/pull/6050).

**Pull request merged**